symfony/client/src/orangehrmAdminPlugin/pages/payGrade/SavePayCurrency.vue

```diff
@@ -0,0 +1,148 @@
+<!--
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+<template>
+  <div class="orangehrm-card-container">
+    <oxd-text tag="h6" class="orangehrm-main-title">Add Currency</oxd-text>
+    <oxd-divider />
+    <oxd-form :loading="isLoading" @submitValid="onSave">
+      <oxd-form-row>
+        <oxd-grid :cols="1" class="orangehrm-full-width-grid">
+          <oxd-grid-item>
+            <oxd-input-field
+              type="dropdown"
```

Comment:
Consider using the [new dropdown](https://rajithakumara-oxd.netlify.app/?path=/story/example-selectinput--default&globals=measureEnabled:false). It has the same API, and the v-model will return selected object instead of array. `<oxd-select-input>` dont have async function support. use `beforeMount()` to pre-fetch currencies.

In template
`<oxd-input-field type="select"  label="Currency" :options="currencies" v-model="payCurrency.currencyId" :rules="rules.currencyId" :disabled="!hasUpdatePermissions" required />`

In Script
```
  data() {
    return {
      currencies: [],
    };
  },

  beforeMount() {
    this.isLoading = true;
    this.http
      .request({
        method: 'GET',
        url: `api/v2/admin/pay-grades/${this.payGradeId}/currencies/allowed`,
        params: {
          limit: 0,
        },
      })
      .then(response => {
        const {data} = response.data;
        this.currencies = data.map(item => {
          return {
            id: item.id,
            label: item.id + ' - ' + item.name,
          };
        });
      })
      .finally(() => {
        this.isLoading = false;
      });
  },
```

Reply:
Done